src/libsyncengine/update_detection/file_system_observer/remotefilesystemobserverworker.cpp

```diff
@@ -127,7 +109,7 @@ ExitCode RemoteFileSystemObserverWorker::generateInitialSnapshot() {
         invalidateSnapshot();
         LOG_SYNCPAL_WARN(_logger, "Remote snapshot generation stopped or failed after: " << elapsedSeconds.count() << "s");
 
-        if (exitCode == ExitCodeNetworkError && exitCause == ExitCauseNetworkTimeout) {
+        if (exitCode == ExitCodeNetworkError || exitCause == ExitCauseNetworkTimeout) {
```

Comment:
Is there a case where we got ExitCauseNetworkTimeout with an exitCode != ExitCodeNetworkError? If yes, is it normal? Else we could maybe not test it:
```suggestion
        if (exitCode == ExitCodeNetworkError) {
```

Reply:
My bad, I miss read twice `exitCode`. I will revert that. Thx